daemon/daemon_main.go

```diff
@@ -1328,6 +1331,24 @@ func runDaemon() {
 		}()
 	}
 
+	// This must be before the API is served to prevent endpoints from being
+	// created before we migrate to the new ENI datapath. It also must be
+	// before the health initialization logic which might create the health
+	// endpoint, for the same reasons as being before the API is served. We
+	// want to ensure that this migration logic runs before any endpoint is
+	// being created.
+	if option.Config.IPAM == option.IPAMENI {
+		err, revErr := linuxrouting.MigrateENIDatapath(&interfaceNumberGetter{})
+		switch {
+		case err != nil && revErr != nil:
+			log.WithError(err).Fatal("Failed to migrate ENI datapath. The original datapath failed to be reinstated. Connectivity is not guaranteed.")
+		case err != nil && revErr == nil:
+			log.WithError(err).Error("Failed to migrate ENI datapath. The original datapath has been reinstated. Connectivity has not been affected.")
```

Comment:
```suggestion
			log.WithError(err).Warning("Failed to migrate ENI datapath. The original datapath has been reinstated. Connectivity has not been affected.")
```

I think we're trying to hint that something is off, but the cluster should otherwise continue most likely unaffected (other than the possibility of hitting the bug that this PR is trying to resolve). Seems like Warning is enough severity for that kind of announcement to the user.